__tests__/server/devHolocronCDN.spec.js

```diff
@@ -109,51 +112,38 @@ describe('devHolocronCDN', () => {
     let oneAppDevCdnMiddleware;
     let devHolocronCDN;
 
-    jest.resetAllMocks();
-
     beforeEach(() => {
       jest.resetModules();
-      jest.mock('cors', () => jest.fn());
-      jest.mock('@americanexpress/one-app-dev-cdn', () => jest.fn());
       const cors = require('cors');
       const oneAppDevCdn = require('@americanexpress/one-app-dev-cdn');
       corsMiddleware = jest.fn((req, res, next) => next());
-      oneAppDevCdnMiddleware = jest.fn((req, res, next) => next());
       cors.mockReturnValue(corsMiddleware);
+      oneAppDevCdnMiddleware = jest.fn((req, res, next) => next());
       oneAppDevCdn.mockReturnValue(oneAppDevCdnMiddleware);
       devHolocronCDN = require('../../src/server/devHolocronCDN').default;
     });
 
-    it('should hit the cors middleware first', (done) => {
+    it('should hit the cors middleware first', async () => {
       corsMiddleware.mockImplementationOnce((req, res) => res.sendStatus(204));
-      request(devHolocronCDN)
-        .get('/static/anything.json')
-        .end((error, { status }) => {
-          expect(corsMiddleware).toHaveBeenCalledTimes(1);
-          expect(oneAppDevCdnMiddleware).not.toHaveBeenCalled();
-          expect(status).toBe(204);
-          done();
-        });
+      const resp = await request(devHolocronCDN)
+        .get('/static/anything.json');
+      expect(corsMiddleware).toHaveBeenCalledTimes(1);
+      expect(oneAppDevCdnMiddleware).not.toHaveBeenCalled();
+      expect(resp.status).toBe(204);
```

Comment:
since this is async we should add an `expect.assertions`

Reply:
not required due to async await pattern

Reply:
not tracking why this shouldn't be required for all a sync tests

Reply:
expect.assertions is used to ensure that X amount of expect are called in scenarios where its possible for the block to finish without calling the expect. This was very important when unhandled promises did not throw an error. 


in this case it used to be possible for the function to return without calling the expect. if the promise rejected this could silently finish without calling the expect which is why we there is the `expect.assertions`. 
```
it('has expect in promise chain', () => {
  somePromise().then((result) => {
    expect(result).toBe(true)
  })
})
```

Now however if `somePromise` fails it will throw 

```
[UnhandledPromiseRejection: This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). The promise rejected with the reason "error".] {
  code: 'ERR_UNHANDLED_REJECTION'
}
```

Reply:
`expect.assertions` is still relevant for conditional logic or where you are using a try catch block

```
it('in try catch block', () => {
    try {
      const wasSafe = possibleKaboom()
      expect(wasSafe).toBe(true)
    } catch {
      // if the above block throws an error this test can fail silently 
      // which is why we need expect.assertions.
    }
  })
```

```
it('conditional tests', () => {
    const itDid = doSomething()
    if(itDid) {
      expect(itDid).toBe(true)
    } else {
      // this is a bad test but if for some reason you need this pattern
      // then expect.assertions is required here
    }
  })
  ```